glue/qt/widgets/histogram_widget.py

```diff
@@ -193,6 +183,21 @@ def component(self, component):
 
     @defer_draw
     def _set_attribute_from_combo(self, *args):
+        if self.component is not None:
+            for d in self._data:
+                try:
+                    component = d.get_component(self.component)
+                except:
+                    continue
+                else:
+                    break
```

Comment:
When ComponentIDs are present in 2 datasets, it's because those two components have been linked together and are conceptually equivalent. The intent of this codeblock is to check if a component is categorical. There's technically nothing stopping you from linking a categorical component with a non-categorical one, but I don't think this would ever legitimately come up in practice. So I think the behavior here (grabbing the first component you find), should be reasonably well defined.